css-shared-element-transitions/index.bs

```diff
@@ -623,106 +703,47 @@ The {{SameDocumentTransition/finished}} [=getter steps=] are to return [=this's=
-      });
-
-      // When the promise returned by prepare() resolves, all pseudo-elements
-      // for this transition have been generated. They can now be accessed
-      // in script to set up custom animations.
-      document.documentElement.animate(
-        keyframes,
-        {...animationOptions,
-         pseudoElement: "::page-transition-container(message)",
-        }
-      );
+                    Note: This happens if |transition| was [=skip the page transition|skipped=] before this point.
 
-      // When the finished promise resolves, that means the transition is
-      // finished (either completed successfully or abandoned).
-      await transition.finished;
-    }
-    </pre>
+                1. [=Skip the page transition=] |transition| with |r|.
 </div>
 
-## Abandon the page transition ## {#abandon-the-page-transition-algorithm}
+## Skip the page transition ## {#skip-the-page-transition-algorithm}
 
 <div algorithm>
-    To <dfn>abandon the page transition</dfn> for {{SameDocumentTransition}} |transition|
-    with |error|:
+    To <dfn>skip the page transition</dfn> for {{DOMTransition}} |transition| with reason |reason|:
 
     1. Let |document| be |transition|'s [=relevant global object's=] [=associated document=].
 
-    1. If |transition|'s [=SameDocumentTransition/phase=] is not "`idle`", then:
+    1. [=Assert=]: |document|'s [=document/active DOM transition=] is |transition|.
+
+    1. [=Assert=]: |transition|'s [=DOMTransition/phase=] is not "`done`".
+
+    1. If |transition|'s [=DOMTransition/phase=] is [=phases/before=] "`dom-update-callback-called`", then [=call the DOM update callback=] of |transition|.
 
-        1. Stop suppressing rendering opportunities |document|, if currently suppressed.
+    1. Set [=document/transition suppressing rendering=] to false.
 
-            Issue: "suppressing rendering opportunities" needs to be linked/defined.
+    1. If |transition|'s [=DOMTransition/phase=] is equal to or [=phases/after=] "`animating`", then:
 
         1. Remove all associated [=page-transition pseudo-elements=] from |document|.
 
             Issue: There needs to be a definition/link for "remove".
 
             Issue: There needs to be a definition/link for "associated".
 
-    1. Set |transition|'s [=SameDocumentTransition/phase=] to "`finished`".
+    1. Set |transition|'s [=DOMTransition/phase=] to "`done`".
 
-    1. If |document|'s [=document/pending same document transition outgoing capture=] is |transition|,
-        then set |document|'s [=document/pending same document transition outgoing capture=] to null.
+    1. Set |document|'s [=document/active DOM transition=] to null.
 
-    1. If |document|'s [=document/running same document transition=] is |transition|,
-        then set |document|'s [=document/running same document transition=] to null.
+    1. [=Reject=] |transition|'s [=DOMTransition/ready promise=] with |reason|.
```

Comment:
Would we still do this if this promise was already resolved? If we skip during the animation phase.

Reply:
Rejecting (and resolving) is a no-op if the promise has already rejected/resolved, but yeah, it'll probably be clearer to be explicit about that, or at least add a note.